### PR TITLE
Hotfix: format notification email date & time by user locale

### PR DESCRIPTION
**Pull request by samanehsan**
<h3>Purpose</h3>
At the moment, the times in notification emails are only shown in 24-hour time. Also, the date is always shown as the month before the day: `17:30 on Tuesday, March 3 EST`. Show the time/date as it would be displayed in the user's locale. For example, a user in Charlottesville would see `11:05 AM on Tuesday, March 3, 2015` but a user in Berlin would see `17:05 on Dienstag, 3. März 2015`. 

<h3>Changes</h3>
Get user locale data from `window.navigator.language` or `window.navigator.userLanguage` (for IE), store on the user model and use to format date and time with the Babel library.

### CODE REVIEW

**Comment by sloria** [comment]
@samanehsan Since this isn't a critical fix and we'll want to do thorough QA on this, can you please resend this as a feature branch off of develop?

File: website/notifications/emails.py [hunk 1]
```diff
@@ -166,7 +165,16 @@ def get_settings_url(uid, user):
...
+
+    try:
+        user_locale = Locale(user.locale)
+    except core.UnknownLocaleError:
+        user_locale = 'en'
+
+    formatted_date = dates.format_date(timestamp, format='full', locale=user_locale)
+    formatted_time = dates.format_time(timestamp, format='short', tzinfo=user_timezone, locale=user_locale)
+
+    return formatted_time + ' on ' + formatted_date
```
Comment (sloria) [inline website/notifications/emails.py]: Prefer `str.format` to string addition.

File: website/static/js/pages/dashboard-page.js [hunk 2]
```diff
@@ -47,18 +47,22 @@ request.fail(function(xhr, textStatus, error) {
     });
 });
 
-var ensureUserTimezone = function(savedTimezone) {
+var ensureUserTimezone = function(savedTimezone, savedLocale) {
     var clientTimezone = jstz.determine().name();
+    var clientLocale = window.navigator.userLanguage || window.navigator.language;
 
-    if (savedTimezone != clientTimezone) {
+    if (savedTimezone != clientTimezone || savedLocale != clientLocale) {
```
Comment (sloria) [inline website/static/js/pages/dashboard-page.js]: Use `!==` for string comparison.

**Comment by sloria** [comment]
Pass finished. Once this is moved over to develop and those minor changes are made, this should be good to merge. :wolf:

**Comment by samanehsan** [comment]
Done responding to first pass! I'll send a PR to the develop branch and close this one. :snowflake: :snowman:

**Pull request closed without merging**